src/storage/storage.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    Injectable,
+    InternalServerErrorException,
+    Logger,
+    LoggerService,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+    ReadStream,
+    WriteStream,
+    createReadStream,
+    createWriteStream,
+} from 'fs';
+import { readdir } from 'fs/promises';
+import { join } from 'path';
+import { v4 as uuid } from 'uuid';
+import * as fs from 'fs';
+
+@Injectable()
+export class StorageService {
+    constructor(private readonly configService: ConfigService) {}
+
+    async saveFile(file: Express.Multer.File): Promise<string> {
+        const storage: string | undefined =
+            this.configService.get<string>('STORAGE');
+
+        if (!storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+
+        const fileId: string = uuid();
+        const fileExtension: string = file.mimetype.includes('html')
+            ? 'html'
+            : 'xml';
+        const writeStream: WriteStream = createWriteStream(
+            join(process.cwd(), `${storage}/${fileId}.${fileExtension}`),
+        );
+        writeStream.write(file);
+
+        return fileId;
+    }
+
+    async readFile(name: string): Promise<ReadStream> {
+        const storage: string | undefined =
+            this.configService.get<string>('STORAGE');
+
+        if (!storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+
+        let filePath: string = '';
+        const dir: string = join(process.cwd(), `${storage}`);
+        const files = await readdir(dir);
+
+        for (const file of files) {
+            if (file.includes(name)) {
+                filePath = join(dir, name);
+                break;
+            }
+        }
+
+        if (filePath === '') {
+            throw new Error('File does not exist');
+        }
+
+        return new Promise((resolve, reject) => {
+            const readStream = fs.createReadStream(filePath);
+            readStream.on('error', (error) => reject(error));
+            readStream.on('end', () => resolve(readStream));
+        });
```

Comment:
Wydaje mi się, że jeśli na "end" mamy zwrotkę to już można zwrócić cały odczytany plik a nie tylko stream? Co chcemy robić dalej z plikiem? Jeśli chcemy wstawić cały plik (templatkę) do szablonu maila aby podmienić w nim $zmienne na treść kontekstu, to raczej potrzebujemy cały plik w postaci stringa a nie stream? Chyba, że chcemy tu zwracać stream a to, o czym piszę robić w innej metodzie. Wtedy nie zwracamy nic na "end" ale:
```
function readMyFileAsStream(filePath: string): NodeJS.ReadableStream {
  return createReadStream(filePath);
}
```

Reply:
Tutaj mogę się po części zgodzić. Generalnie jeżeli chodzi o readFile to miałem różne koncepcje. Na początku zwracałem string, potem zmieniłem na ReadStream. Został ReadStream, aby wykorzystać potem StreamableFile w kontrolerze do zwrotu pliku. Ale tak jak napisałeś, będziemy to też wykorzystywać do podstawienia pod template i wstrzyknięcia kontekstu kilenta. 
Także wydaje mi się, że to poprawię i zwróce string. Będzie to bardzie uniwersalne.

Reply:
moze zostac stream - pamietajcie aby minilamizowac zagniezdzony kod wiec to co ty chcesz zrobic mozna przeniesc do innej metody ktora bedzie dzialac na streamie juz